mypy/checker.py

```diff
@@ -2612,9 +2613,17 @@ def check_member_assignment(self, instance_type: Type, attribute_type: Type,
         care about interaction between binder and __set__().
         """
         # Descriptors don't participate in class-attribute access
-        if ((isinstance(instance_type, FunctionLike) and instance_type.is_type_obj()) or
-                isinstance(instance_type, TypeType)):
-            rvalue_type = self.check_simple_assignment(attribute_type, rvalue, context)
+        if (isinstance(instance_type, CallableType) and instance_type.is_type_obj() and
+                isinstance(instance_type.ret_type, Instance)):
```

Comment:
Named tuple class object have a `TupleType` return type, you can "convert" it to instance by using its fallback.